File: apps/agent/src/archetypes.py
"""Archetype catalog loader for Counterpart V4.

Single source of truth: ../../data/archetypes.json
"""
from __future__ import annotations

import json
from dataclasses import dataclass, field
from pathlib import Path

_DATA_FILE = Path(__file__).parents[3] / "data" / "archetypes.json"


@dataclass
class Archetype:
    nombre: str
    branch_level_1: str
    sub_branch: str | None
    descripcion: str
    disc_quadrant: str
    voss_type: str
    ekman_emotion: str
    ekman_signal: str
    ocean_base: dict
    primary_fear: str
    sparring_style: str
    entry_point: str
    blind_spot: str
    close_signal: str
    key_lever: str
    batna_awareness: str
    institutional_context_prior: dict
    demo_responses: list[dict] = field(default_factory=list)
    keywords: list[str] = field(default_factory=list)


def _load() -> list[Archetype]:
    raw = json.loads(_DATA_FILE.read_text(encoding="utf-8"))
    result = []
    for a in raw["archetypes"]:
        result.append(
            Archetype(
                nombre=a["nombre"],
                branch_level_1=a["branch_level_1"],
                sub_branch=a.get("sub_branch"),
                descripcion=a["descripcion"],
                disc_quadrant=a["disc_quadrant"],
                voss_type=a["voss_type"],
                ekman_emotion=a["ekman_emotion"],
                ekman_signal=a["ekman_signal"],
                ocean_base=a["ocean_base"],
                primary_fear=a["primary_fear"],
                sparring_style=a["sparring_style"],
                entry_point=a["entry_point"],
                blind_spot=a["blind_spot"],
                close_signal=a["close_signal"],
                key_lever=a["key_lever"],
                batna_awareness=a["batna_awareness"],
                institutional_context_prior=a["institutional_context_prior"],
                demo_responses=a.get("demo_responses", []),
                keywords=a.get("keywords", []),
            )
        )
    return result


ARCHETYPES: list[Archetype] = _load()
ARCHETYPE_NAMES: list[str] = [a.nombre for a in ARCHETYPES]
ARCHETYPE_NAMES_CSV: str = ", ".join(ARCHETYPE_NAMES)

_BY_NAME: dict[str, Archetype] = {a.nombre: a for a in ARCHETYPES}


def archetype_by_name(nombre: str) -> Archetype | None:
    return _BY_NAME.get(nombre)


def archetype_catalog_text() -> str:
    lines = []
    for a in ARCHETYPES:
        lines.append(
            f"- {a.nombre} (rama: {a.branch_level_1}, DISC: {a.disc_quadrant}, "
            f"Voss: {a.voss_type}) — {a.descripcion}"
        )
    return "\n".join(lines)

File: apps/mcp/src/data/archetypes.ts
import { createRequire } from "module";
import { z } from "zod";
import { fileURLToPath } from "url";
import path from "path";

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const require = createRequire(import.meta.url);

const rawData = require(path.join(__dirname, "../../../../data/archetypes.json"));

const ArchetypeSchema = z.object({
  nombre: z.string(),
  branch_level_1: z.string(),
  sub_branch: z.string().nullable().optional(),
  descripcion: z.string(),
  disc_quadrant: z.string(),
  voss_type: z.string(),
  ekman_emotion: z.string(),
  ekman_signal: z.string(),
  ocean_base: z.record(z.number()),
  primary_fear: z.string(),
  sparring_style: z.string(),
  entry_point: z.string(),
  blind_spot: z.string(),
  close_signal: z.string(),
  key_lever: z.string(),
  batna_awareness: z.string(),
  institutional_context_prior: z.record(z.number()),
  demo_responses: z.array(z.any()).optional().default([]),
  keywords: z.array(z.string()).optional().default([]),
});

export type Archetype = z.infer<typeof ArchetypeSchema>;

export const ARCHETYPES: Archetype[] = rawData.archetypes.map((a: unknown) =>
  ArchetypeSchema.parse(a)
);

export const ARCHETYPE_NAMES = ARCHETYPES.map((a) => a.nombre);

export function archetypeByName(nombre: string): Archetype | undefined {
  return ARCHETYPES.find((a) => a.nombre === nombre);
}
